utils/fixed_decimal/src/lib.rs

```diff
@@ -91,6 +93,9 @@ pub enum Error {
     /// 123 (or 123.0) must be used.
     #[displaydoc("Failed to parse the input string")]
     Syntax,
+    /// An exponent with fractional digits is passed when constructing a [`ScientificDecimal`].
+    #[displaydoc("Exponent must not have fractional digits")]
+    IntegerExponent,
```

Comment:
Nit: It doesn't make sense to split IntegerExponent into its own enum separate from Syntax unless you do the same for other specific types of syntax errors.

Reply:
Yeah, this has an uncomfortable « true/false/file not found » æsthetic.
I had done that because from_parts isn’t parsing (it takes FixedDecimals that are already structured), so this technically isn’t a syntax error; with `FixedInteger` this should cease to exist.